### PR TITLE
Fix a panic on startup with outstanding invite

### PR DESCRIPTION
**Pull request by DMRobertson**
if the proxy is not aware of anyone else joined to the room.

### CODE REVIEW

File: state/storage.go [hunk 1]
```diff
@@ -206,6 +209,8 @@ func (s *Storage) MetadataForAllRooms(txn *sqlx.Tx, tempTableName string, result
 		metadata.LatestEventsByType[parsed.Get("type").Str] = eventMetadata
 		// it's possible the latest event is a brand new room not caught by the first SELECT for joined
 		// rooms e.g when you're invited to a room so we need to make sure to set the metadata again here
+		// TODO: is this accurate now that we explicitly call NewRoomMetadata above
+		//       when handling invites?
```
Comment (kegsay) [inline state/storage.go:213]: Is what accurate? The room ID? It should be if we're setting it in NewRoomMetadata?

**Review by kegsay — APPROVED** [review]
Good find, thanks!

File: state/storage.go (hunk 1 above)
Comment (DMRobertson) [reply state/storage.go:213]: Sorry "this" means "the comment above". I.e. _is_ it possible for us to arrive at line 202 now?

My thinking is "maybe not, but I don't have the time to investigate further and it's not broken as-is".

Comment (DMRobertson) [inline state/storage.go:213]:
```suggestion
		// TODO: is the comment above now that we explicitly call NewRoomMetadata above
		//       when handling invites?
```

**Pull request merged**